hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -198,7 +198,8 @@ public class HoodieTableConfig extends HoodieConfig {
   
   public static final ConfigProperty<RecordMergeMode> RECORD_MERGE_MODE = ConfigProperty
       .key("hoodie.record.merge.mode")
-      .defaultValue(RecordMergeMode.EVENT_TIME_ORDERING)
+      .defaultValue((RecordMergeMode) null,
+          "COMMIT_TIME_ORDERING if precombine is not set; EVENT_TIME_ORDERING if precombine is set")
```

Comment:
Can we simplify the logic in `HoodieTableConfig.inferCorrectMergingBehavior` ? It looks way too complicated.

Reply:
Makes sense.  I'll take a stab.

Reply:
@danny0405 I've taken a stab to simplify it. lmk if it looks better and easier to understand.